Handler/Cancel.hs

```diff
@@ -0,0 +1,22 @@
+module Handler.Cancel where
+
+import Import
+import Stripe
+
+postCancelR :: Handler Html
+postCancelR = do
+    plan <- runDB $ getBy404 $ UniquePlan freePlanId
+    Entity userId user <- requireAuth
```

Comment:
The left hand side is pattern matching. `requireAuth` returns an `Entity User`, which is the `UserId` and `User` together in one data type. This just pattern matches it into separate variables because they're used separately after.